### PR TITLE
Integrate VCID Routing

### PR DESCRIPTION
**Pull request by Mejiro-McQueen**
Added TaggedFrame class.
Other minor changes to hammer VCID router into place.

### CODE REVIEW

**Review by cSchubes — APPROVED** [review]

File: ait/dsn/plugins/vcid_routing.py [hunk 1]
```diff
@@ -56,19 +62,25 @@ def __init__(self, inputs=None, outputs=None, zmq_args=None, routing_table=None,
...
         :param input_data: AOS frame as bytes
         :type input_data: bytes, bytearray 
         '''
-        frame_vcid = self.get_frame_vcid(input_data)
+        #frame_vcid = self.get_frame_vcid(input_data)
+        frame_vcid = tagged_frame.vcid
         if frame_vcid in self.routing_table_object:
             topics = self.routing_table_object[frame_vcid]
+            self.vcid_counter[frame_vcid] += 1
+            tagged_frame.vcid_counter = self.vcid_counter[frame_vcid]
```
Comment (cSchubes) [inline ait/dsn/plugins/vcid_routing.py:79]: should this be `tagged_frame.counter = self.vcid_counter[frame_vcid]`?

**Review by cSchubes — CHANGES_REQUESTED** [review]
I don't see where https://jira.jpl.nasa.gov/browse/SRGDS-100 is addressed, here or in https://github.jpl.nasa.gov/SunRISE-Ops/SunRISE-AIT/pull/51. Apologies if I'm just missing something.

**Comment by Mejiro-McQueen** [comment]
> I don't see where https://jira.jpl.nasa.gov/browse/SRGDS-100 is addressed, here or in https://github.jpl.nasa.gov/SunRISE-Ops/SunRISE-AIT/pull/51. Apologies if I'm just missing something.

## BLUF: We need Frame Processors, and integrating the VCID Router and AOS FEC Checker updates are a prerequisite for writing them. 

There's two parts to that ticket, so it's confusing to reference. 
This branch addresses the AOS FEC Checker Updates portion. 
The Frame Archive portion should have been its own ticket with Frame Processors as a prerequisite (I'll handle it). 

## AOS FEC Checker Updates Portion
TL;DR: All Frames make it past the VCID router and into a Frame Processor. Whether the Frame is good enough for processing is the Processor's decision.

### Mangled VCID (Corruption is necessarily also true)
A bad VCID won't match any of our known VCIDS.
Our VCID routing table routes into the Frame Processors for an explicit VCID.
Since we didn't match any Frame Processor route, or any route in our routing table for that matter, the mangled frame will be sent to the default VCID route, which is a FrameProcessor called dev_null, which as you can imagine, just tosses it out into the void.

All frames with invalid VCID pass through, but the route is separate from good frames.

### Corrupt Frames, VCID is ok
In the case of say a corrupt frame with a matching VCID, say routed to RealTimeTLM/StoredTLM Processors, we add a check just before attempting to do anything with it. 

So all frames with valid VCID pass through, but the Frame Processor can decide whether it wants to use a corrupt frame or not.

# Frame Archival (Should be its own ticket or sub ticket)
Frame Archiver should really be a Frame Processor.
We can rename that plugin so that it can collect from an arbitrary PUB/SUB topic, which would be really useful for testing.

We can specify which VCIDs we are interesting in archiving by modifying the routing table.

## Don't Archive Idle Frames and Separate Corrupt Frames
Frame Archive Processor won't have a route from Idle Frames
Frame Archive Processor will check the corrupt? flag when deciding where to write the frame.

## Archive as JSON 
Should be addressed as part of a Frame Archive Processor ticket.

**Review by jbust23 — APPROVED** [review]
